gramps/gen/const.py

```diff
@@ -99,10 +99,13 @@
     HOME_DIR = os.path.join(USER_HOME, "gramps")
 elif "USERPROFILE" in os.environ:
```

Comment:
The code relies on the existence of an environment variable to infer the platform and run platform specific code.
Is it better to directly query the platform using `sys.platform == "win32"` rather than inferring it? This would cover the scenario of a Linux or mac user not having `GRAMPSHOME` set and having `USERPROFILE` set. Maybe unlikely, but it won't be easy to debug if it does occur.

This might be better handled in a separate PR as it is outside the scope of this PR

Reply:
@stevenyoungs Thanks for the review. It does appear that the USERPROFILE block is intended only for Windows, so your suggestion would make it explicit. OTOH, I don't know if Linux/Mac users setting USERPROFILE is supposed to be a supported case. Agree with creating another PR if its decided to change this.

Reply:
> I don't know if Linux/Mac users setting USERPROFILE is supposed to be a supported case.

No. The `USERPROFILE` environment variable is Windows only.